File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/SpellsPerDayTableRowMapper.java
package com.d20charactersheet.framework.dac.dao.sql.rowmapper;

import java.sql.SQLException;

import org.jetbrains.annotations.NotNull;

import com.d20charactersheet.framework.boc.model.SpellsPerDayTable;
import com.d20charactersheet.framework.dac.dao.sql.DataRow;
import com.d20charactersheet.framework.dac.dao.sql.RowMapper;

/**
 * Maps spell per day tables.
 */
public class SpellsPerDayTableRowMapper implements RowMapper {

    @NotNull
    @Override
    public Object mapRow(@NotNull DataRow dataRow) throws SQLException {
        final SpellsPerDayTable spellsPerDayTable = new SpellsPerDayTable();
        spellsPerDayTable.setId(dataRow.getInt(0));
        spellsPerDayTable.setName(dataRow.getString(1));
        return spellsPerDayTable;
    }

}

File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/SpellsPerDayTableRowMapper.kt
package com.d20charactersheet.framework.dac.dao.sql.rowmapper

import com.d20charactersheet.framework.boc.model.SpellsPerDayTable
import com.d20charactersheet.framework.dac.dao.sql.DataRow
import com.d20charactersheet.framework.dac.dao.sql.RowMapper
import java.sql.SQLException

class SpellsPerDayTableRowMapper : RowMapper {

    /**
     * Maps spell per day tables.
     */
    @Throws(SQLException::class)
    override fun mapRow(dataRow: DataRow): Any {
        val spellsPerDayTable = SpellsPerDayTable()
        spellsPerDayTable.id = dataRow.getInt(0)
        spellsPerDayTable.name = dataRow.getString(1)
        return spellsPerDayTable
    }
}